src/main/webapp/script.js

```diff
@@ -11,3 +11,79 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/*
+ * Displays navbar authentication buttons according to login status.
+ */
+window.onload = function fetchLogin() {
+  fetch('/login').then(response => response.json()).then(user => {
+    // If user is logged in, show logout button in navbar.
+    if (user.loggedIn == true) {
+      console.log("logged in");
+
+      // Delete signup button.
+      const signupButtonNavbar = document.getElementById('signup');
+      signupButtonNavbar.innerHTML = '';
+
+      // Create logout button.
+      const logoutButton = document.createElement('button');
+      logoutButton.setAttribute('type', 'button');
+      const logoutUrl = 'window.location.href = \"' + user.authenticationUrl + '\"';
+      logoutButton.setAttribute('onclick', logoutUrl);
+      logoutButton.classList.add('btn');
+      logoutButton.classList.add('btn-outline-success');
+      logoutButton.innerHTML = 'Log Out';
+
+      // Create navbar item to hold logout button.
+      const logoutButtonItem = document.createElement('li');
+      logoutButtonItem.classList.add('nav-item');
+      logoutButtonItem.appendChild(logoutButton);
+    
+      // Append logout button to navbar.
+      const logoutButtonNavbar = document.getElementById('login');
+      logoutButtonNavbar.innerHTML = '';
+      logoutButtonNavbar.appendChild(logoutButtonItem);
+    // If user is logged out, show signup and login buttons in navbar.
+    } else {
+      console.log("logged out");
+
+      // Create signup button.
+      const signupButton = document.createElement('button');
+      signupButton.setAttribute('type', 'button');
+      const signupUrl = 'window.location.href = \"' + user.authenticationUrl + '\"';
+      signupButton.setAttribute('onclick', signupUrl);
+      signupButton.classList.add('btn');
+      signupButton.classList.add('btn-success');
+      signupButton.innerHTML = 'Sign Up';
```

Comment:
Some of this logic seems shared between multiple buttons, consider factoring it out into a helper function.

Reply:
Same with the navbar logic - consider how to refactor this function to share logic while maintaining readability!

Reply:
Wow, looks much cleaner now! Done.